src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextClientMetadata.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Provides metadata about an <see cref="ISpeechToTextClient"/>.</summary>
+public class SpeechToTextClientMetadata
+{
+    /// <summary>Initializes a new instance of the <see cref="SpeechToTextClientMetadata"/> class.</summary>
+    /// <param name="providerName">
+    /// The name of the speech to text  provider, if applicable. Where possible, this should map to the
+    /// appropriate name defined in the OpenTelemetry Semantic Conventions for Generative AI systems.
+    /// </param>
+    /// <param name="providerUri">The URL for accessing the speech to text  provider, if applicable.</param>
+    /// <param name="modelId">The ID of the speech to text  model used, if applicable.</param>
+    public SpeechToTextClientMetadata(string? providerName = null, Uri? providerUri = null, string? modelId = null)
```

Comment:
Is there any other common metadata that all known providers support?

Reply:
I could not think of anything specific yet.